app/Repositories/PostRepository.php

```diff
@@ -87,7 +88,17 @@ public function create(array $data, $signupId, $authenticatedUserRole = null)
                 'campaign_id' => $signup->campaign_id,
                 'post_type' => $data['type'],
                 'name' => $data['action'],
-            ])->firstOrFail();
+            ])->first();
+
+            if (! $action) {
+                info('action_not_found', [
+                    'campaign_id' => $signup->campaign_id,
+                    'post_type' => $data['type'],
+                    'name' => $data['action'],
+                ]);
+
+                throw new ValidationException::withMessages(array_fill_keys(['campaign_id', 'post_type', 'name']), 'An action with the given fields does not exist.');
+            }
```

Comment:
@DFurnes I'm getting this error but I think everything looks ok to me? 
```
Unexpected 'withMessages' (T_STRING), expecting variable (T_VARIABLE) or '$' on line 100.
```

Reply:
Oh oops, since we're calling a static method on that class (which returns an instance), we can just `throw ValidationException::withMessages(…)`.

Reply:
ah that worked! Thank you!